invoices/invoiceregistry.go

```diff
@@ -447,7 +509,22 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 		expiry:               expiry,
 		currentHeight:        currentHeight,
 		finalCltvRejectDelta: i.cfg.FinalCltvRejectDelta,
-		customRecords:        payload.CustomRecords(),
+		customRecords:        customRecords,
+	}
+
+	// Process key send if present. Do this outside of the lock, because
+	// AddInvoice obtains its own lock. This is no problem, because the
+	// operation is idempotent.
+	if i.cfg.AcceptKeySend {
+		err := i.processKeySend(updateCtx, rHash)
+		if err != nil {
+			debugLog(fmt.Sprintf("key send error: %v", err))
+
+			return &HodlEvent{
```

Comment:
So we'll just hold onto it if the pre-image doesn't match the payment hash? I think instead, we want to cancel it like any other HTLC.

Reply:
Or is this meant to allow a caller using the HODL invoice calls to _manually_ settle this?

Reply:
No, returning a `HodlEvent` without a preimage is an immediate cancel. So it behaves like you expect.